src/components/portfolios/PortfolioDetailTemplate.jsx

```diff
@@ -0,0 +1,216 @@
+import CircularProgress from "@mui/material/CircularProgress";
+import {
+  child,
+  get,
+  getDatabase,
+  ref,
+  serverTimestamp,
+  set,
+} from "firebase/database";
+import { useSetAtom } from "jotai";
+import React, { useState } from "react";
+import { useSelector } from "react-redux";
+import { useNavigate } from "react-router-dom";
+import { createChatRoom } from "../../apis/chat";
+import { ReactComponent as RightArrow } from "../../assets/right-01.svg";
+import "../../firebase";
+import { paymentAtom } from "../../store";
+import { comma } from "../../utils/convert";
+import Button from "../common/atoms/Button";
+import DivideBar from "../common/atoms/DivideBar";
+import PaymentBottomSheet from "../profile/PaymentBottomSheet";
+import DescriptionRow from "./DescriptionRow";
+import HistoryBottomSheet from "./HistoryBottomSheet";
+import PortfolioCarousel from "./PortfolioCarousel";
+import PriceInfoRow from "./PriceInfoRow";
+
+const PortfolioDetailTemplate = ({ portfolio }) => {
+  const { userInfo } = useSelector((state) => state.user);
+  const [paymentBottomSheetOpen, setPaymentBottomSheetOpen] = useState(false);
+  const [historyBottomSheetOpen, setHistoryBottomSheetOpen] = useState(false);
+  const setPayment = useSetAtom(paymentAtom);
+  const navigate = useNavigate();
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const handleOnCreateChatRoom = async () => {
+    setIsSubmitting(true);
+    const db = getDatabase();
+    const userRef = ref(db, `users/${userInfo.userId}`);
+    const counterUserRef = ref(db, `users/${portfolio.userId}`);
+    try {
+      const snapshot = await get(userRef);
+      const data = snapshot.val();
+      if (data && data.chatRooms) {
+        const chatIds = Object.keys(data.chatRooms);
+        const existChatRoom = chatIds.find((chatId) => {
+          const chatRoom = data.chatRooms[chatId];
+          return chatRoom.counterId === portfolio.userId;
+        });
+        if (existChatRoom) {
+          // console.log("이미 채팅방이 존재합니다.");
+          navigate(`/chat/${existChatRoom}`);
+          return;
+        }
+      }
+      // 채팅방이 없을 경우 새로 만들어준다.
+      const res = await createChatRoom(portfolio.userId);
+      const { chatId } = res.response;
```

Comment:
```suggestion
      const { response: {chatId} } = await createChatRoom(portfolio.userId);
```
이렇게 축약하시는 것도 좋을 것 같습니다!